opamp-client/src/main/java/io/opentelemetry/opamp/client/OpampClientBuilder.java

```diff
@@ -30,27 +34,57 @@
 
 /** Builds an {@link OpampClient} instance. */
 public final class OpampClientBuilder {
+  private static final String DEFAULT_ENDPOINT_URL = "http://localhost:4320/v1/opamp";
+  private static final URI DEFAULT_ENDPOINT = URI.create(DEFAULT_ENDPOINT_URL);
+
   private final Map<String, AnyValue> identifyingAttributes = new HashMap<>();
   private final Map<String, AnyValue> nonIdentifyingAttributes = new HashMap<>();
   private long capabilities = 0;
-  private RequestService service =
-      HttpRequestService.create(OkHttpSender.create("http://localhost:4320/v1/opamp"));
+  private Function<URI, RequestService> serviceFactory =
```

Comment:
It seems a little unconventional in Java to use a function for this kind of thing. I'd prefer that we just stick with using a method...which I think can even be static.

Reply:
Currently we allow users replace this factory function, so they could customize things. Ideally eventually this will go away when we can provide users an option to customize the okhttp client without having to mess with the RequestService or HttpSender implementations.